### ISSUE TITLE
Z axis as stepper motor PU/PD

### ISSUE THREAD

**Issue by Marmadoc** [svenhb/GRBL-Plotter#400]
I try to understand how Z Axis Pen Up/Down works
I have stepper 24 BYJ 48 12V.
When I press Z jogging 10z motor moves a little next stops and start moves again and finally stops. Stops jogging on mouse is OFF
on picture is my setup for Z axis
![GRBL Plotter1](https://github.com/svenhb/GRBL-Plotter/assets/69551886/3c5fc8cf-2bb6-4fd0-8f1a-4afdf11da27e)
My mechanical design is 10mm up and 10m down for pen holder.
I calculated that 1mm it is 102 steps (#102 GRBL)
How to use setup to get movement different than 2mm - I tried 20mm but not move that much.
Thx in advance

**Comment by svenhb** [comment]
If you import a graphic, each path starts with a "pen down" to start drawing and ends with a "pen up": Here you can define the absolute positions for your Z axis, If you want full range of 20 mm you may set Z-safe = +10 (above "workpiece" = paper) and Z-engrave = -10 mm (below workpiece-surface):
![image](https://github.com/svenhb/GRBL-Plotter/assets/19973251/e6293bdc-b7fd-487e-bd0a-f0bec16e8190)

Be sure your $102 is correct: if you press the upper "5" does it move 5 mm upwards, and 5 mm downwards if you press the lower "5". If not, you need to check your $102 setup (using micro-steps?) or the motor is too weak and needs more current from the motor driver.
![image](https://github.com/svenhb/GRBL-Plotter/assets/19973251/d8e4b1ef-bbb2-419e-9843-38543c611797)

**Comment by Marmadoc** [comment]
Thank you svenhb. My pen holder and motor is on picture.
Evry 180 deg it moves 10mm (10mm up and 10mm down).
180deg of motor shaft it is 1019 steps, so 1mm it 102 steps. (#102)
I still do not understand why when I click 10Z motor moves a little next stops and start moves again and finally stops.
the same when (Move Pen to upper position) $J=G90 F5000 Z#GMAZ
and (Move Pen to lower position) G90 G0 Z#GMIZ
![Pen](https://github.com/svenhb/GRBL-Plotter/assets/69551886/536bcfdd-85f6-43ed-a782-4cbfac5deb71)
Thx 
Adam

**Comment by svenhb** [comment]
>motor moves a little next stops and start moves again

I assume you watch the pen-holder or z-axis but not the motor axis itself...

I assume, the motor is fixed with the two platic screws and the black disk pushes the red lever?
![image](https://github.com/svenhb/GRBL-Plotter/assets/19973251/c03c9a7e-754d-4ef9-8cee-8ca40cc298cd)

I think friction at the guide axes and the "non-perfect" shape of the disc is the problem.
Also the relation between motor-turn and z-movement is probably not linear

**Comment by svenhb** [comment]
At the red marked positions, the lever won't move, because the radius doesn't change:
![image](https://github.com/svenhb/GRBL-Plotter/assets/19973251/da8f32f8-5ed8-4bd6-bedf-cd430ca2ef7a)

Also the connection between disc and motor axis doesn't look solid

**Comment by Marmadoc** [comment]
It is motor only - one click on Z 10.
https://github.com/svenhb/GRBL-Plotter/assets/69551886/c7b033ee-43a9-4f58-81ee-eedc5057d319
Why it stops in the middle?

**Comment by svenhb** [comment]
Perhaps the gear is broken, can you check with a motor without gear?

**Comment by Marmadoc** [comment]
I found what is wrong. Your F for Z10 is 1000, For Z 5, 2, 1 is 500. When I changed F to 500 for Z 10 - all works OK. It is for 24 BYJ 48 12V, and not for NEMA17 i.e. 24 BYJ 48 12V step angle is ~5.625 before reduction gears. F1000 is too fast for this motor.
Nema17 step angle is 1.9 deg - and this motor can handle speed F1000. For 24 BYJ 48 12V I calculated max speed ~460.
About mechanical design. I'm sure it is sturdy. Plastic mounting screws are in brass inserts. No motor movement at all.
Z movement non-linear. It is very close to linear, and even it is not shouldn't be any problem. It is not X or Y Axis - where precision is required - but pen movement Up and Down - I do not think it is a problem. 
Bottom line - this system with step motor is much nicer than servo. I'm happy with this.
Only I added limit switch and G code starts with homing cycle ( only Z axis). From this point I know exactly pen holder position.
I would be more happy if my blocks are made from aluminum - not 3D plastic printing, but it is very expensive for one off design.
Thx
Ada

**Comment by Marmadoc** [comment]
Thank You. BTW where is jogging setup? I.e $J=G91 Z5 F500

**Comment by svenhb** [comment]
![image](https://github.com/svenhb/GRBL-Plotter/assets/19973251/b883ea81-94b4-4976-b579-8e041b349b1c)

**Comment by Marmadoc** [comment]
Another hidden gem - Thank You
